test/auth.test.js

```diff
@@ -58,15 +67,48 @@ describe('Auth', () => {
     return client.login('email', 'password')
       .then(userId => expect(userId).toEqual('fake-user-id'));
   });
+});
 
-  it('should return a promise for login with only existing auth data userId', async() => {
-    window.fetch.resolves(mockApiResponse());
-    expect.assertions(1);
+describe('Auth login semantics', () => {
+  const email = 'test_user@domain.com';
+  const password = 'password';
+  let test = new StitchMongoFixture();
+  let th;
+  let client;
 
-    let client = new StitchClient();
-    await client.auth.storage.set(common.USER_AUTH_KEY, mockAuthData());
+  beforeAll(async() => await test.setup());
+  afterAll(async() => await test.teardown());
 
-    return client.login('email', 'password')
-      .then(userId => expect(userId).toEqual('fake-user-id'));
+  beforeEach(async() => {
+    const { apiKey, groupId, serverUrl } = extractTestFixtureDataPoints(test);
+    th = await buildClientTestHarness(apiKey, groupId, serverUrl);
+    await th.configureAnon();
+    client = th.stitchClient;
+  });
+
+  afterEach(async() => await th.cleanup());
+
+  it('should track currently logged in provider type', async() => {
+    await client.login();
+    expect(await client.auth.storage.get(common.USER_LOGGED_IN_PT_KEY)).toEqual('anon');
+    await client.login(email, password);
+    expect(await client.auth.storage.get(common.USER_LOGGED_IN_PT_KEY)).toEqual('userpass');
+  });
+
+  it('should return existing login for relogging "anon"', async() => {
+    const userId = await client.login();
+    expect(userId).toEqual(await client.login());
+  });
+
+  it('should login with a new provider if switching providers', async() => {
```

Comment:
it looks like we sorta already do this in one of the previous tests at L94-95. Maybe we should just nest this `it` block into there?

Reply:
Sure, I don't have a strong opinion here. This test was just more explicit.